src/platform-implementation-js/dom-driver/gmail/gmail-driver.ts

```diff
@@ -827,37 +825,42 @@ class GmailDriver {
     ).takeUntilBy(this.#stopper);
   }
 
-  #setupRouteSubViewDriver(viewName: string): Kefir.Observable<any, unknown> {
+  #setupRouteSubViewDriver(
+    viewName: 'newGmailThreadView',
+  ): Kefir.Observable<GmailThreadView, unknown>;
+  #setupRouteSubViewDriver(
+    viewName: 'newGmailRowListView',
+  ): Kefir.Observable<GmailRowListView, unknown>;
+  #setupRouteSubViewDriver(viewName: string) {
     return this.#routeViewDriverStream.flatMap((gmailRouteView) => {
       return gmailRouteView
         .getEventStream()
-        .filter((event: any) => event.eventName === viewName)
-        .map((event: any) => event.view);
+        .filter((event) => event.eventName === viewName)
+        .map((event) => event.view);
     });
   }
 
   #setupThreadRowViewDriverKefirStream() {
     this.#threadRowViewDriverKefirStream = this.#rowListViewDriverStream
-      .flatMap((rowListViewDriver: GmailRowListView) =>
+      .flatMap((rowListViewDriver) =>
         rowListViewDriver.getRowViewDriverStream(),
       )
       .takeUntilBy(this.#stopper);
   }
 
   #setupToolbarViewDriverStream() {
     this.#toolbarViewDriverLiveSet = toLiveSet(
-      Kefir.merge([
-        this.#rowListViewDriverStream.map(
-          (gmailRowListView: GmailRowListView) =>
-            gmailRowListView.getToolbarView(),
+      Kefir.merge<GmailToolbarView | null | undefined, unknown>([
+        this.#rowListViewDriverStream.map((gmailRowListView) =>
+          gmailRowListView.getToolbarView(),
         ),
         this.getThreadViewDriverStream().map((gmailThreadView) =>
           gmailThreadView.getToolbarView(),
         ),
       ])
-        .filter(Boolean)
+        .filter(isNotNil)
```

Comment:
This should be the only runtime change in this PR. `isNotNil` has a type assertion that feeds into the rest of the method chain.